fix: use image prop from ai-studio.yaml when defined

lstocchi opened the pull request:
### What does this PR do?

It adds support for the new image prop added to ai-studio.yaml file.

### Screenshot / video of UI

N/A

### What issues does this PR fix or reference?

it resolves #161 

### How to test this PR?

1. run any recipe and check that the resulting image have the name defined in its ai-studio. You should delete your local recipe repo as it has to download the new one. If you uses an old repo nothing should change

benoitf commented on packages/backend/src/managers/applicationManager.spec.ts:
```diff
@@ -1358,3 +1358,59 @@ describe('pod detection', async () => {
+    const imageTag = manager.getImageTag(recipe, container);
+    expect(imageTag).equals('quay.io/repo/image:latest');
+  });
+})
```
looks the file is unformatted, don't know if we need pre-commit hooks like in podman-desktop

lstocchi replied:
yes i guess we should add it

lstocchi replied:
opened issue https://github.com/projectatomic/ai-studio/issues/603

sallyom commented:
This will push the end-to-end story forward, making it possible to take the pod.yaml that ai-lab creates and `podman kube play pod.yaml` on any system. 

(as a follow-up, an option to push the ai-lab built images would be really convenient)

jeffmaury commented:
Update to the packaging guide is missing

lstocchi commented:
> Update to the packaging guide is missing

WDYM? I added the image prop description and updated the example -> https://github.com/projectatomic/ai-studio/blob/bdeceea78f90bd8769c71493368320f26b64fce8/PACKAGING-GUIDE.md

jeffmaury commented:
> > Update to the packaging guide is missing
> 
> WDYM? I added the image prop description and updated the example -> https://github.com/projectatomic/ai-studio/blob/bdeceea78f90bd8769c71493368320f26b64fce8/PACKAGING-GUIDE.md

My fault, sorry

jeffmaury reviewed (approved):
LGTM

lstocchi commented:
> (as a follow-up, an option to push the ai-lab built images would be really convenient)

@sallyom Could you please open an issue so we don't forget about it?  Thank you!!
